core/prelude/types/cpp/int.carbon

```diff
@@ -0,0 +1,40 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+// Exceptions. See /LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+package Core library "prelude/types/cpp/int";
+
+import library "prelude/operators";
+import library "prelude/types/int";
+import library "prelude/types/int_literal";
+
+namespace CppCompat;
+
+private fn CppCompat.MakeInt(size: IntLiteral()) -> type = "int.make_type_signed";
+
+// TODO: Add ULong32, LongLong64, ULongLong64.
+class CppCompat.Long32 {
+  adapt MakeInt(32);
```

Comment:
Given that you're importing `prelude/types/int` anyway, perhaps `adapt i32;` and avoid the need to redeclare the builtin?